### PR TITLE
Support integration of IoT Models Repository metadata in preview SDK

### PR DESCRIPTION
**Pull request by digimaun**
- Consumes new service metadata capability outlined [here](https://github.com/Azure/iot-plugandplay-models-tools/wiki/Publishing-Metadata).
  - Metadata for the global endpoint resides [here](https://devicemodels.azure.com/metadata.json).
  - Metadata is optional. Failure in processing repository metadata should not be a terminal failure.
- Simplifies client API surface area:
   - Removes model dependency resolution configuration in the `ModelsRepositoryClientOptions` (and the pattern of the client constructor setting a default resolution option).
   -  Removes `ModelDependencyResolution.TryFromExpanded`. TryFromExpanded becomes an internal processing concept. The metadata of a target repository will indicate if expanded model forms are available.
   - Model dependency resolution configuration becomes scoped to the service operation level. The `GetModels[Async]` service operations have their `dependencyResolution` parameter set to `ModelDependencyResolution.Enabled` by default.
 - Updates to tests and samples documentation reflecting the prior mentioned changes.

### CODE REVIEW

File: sdk/modelsrepository/Azure.IoT.ModelsRepository/src/ModelsRepositoryClient.cs [hunk 1]
```diff
@@ -72,20 +71,20 @@ public ModelsRepositoryClient(Uri repositoryUri, ModelsRepositoryClientOptions o
         /// </returns>
         /// <exception cref="RequestFailedException">Thrown when a resolution failure occurs.</exception>
         /// <param name="dtmi">A well-formed DTDL model Id. For example 'dtmi:com:example:Thermostat;1'.</param>
-        /// <param name="dependencyResolution">A ModelDependencyResolution value to force model resolution behavior.</param>
+        /// <param name="dependencyResolution">A ModelDependencyResolution value to control model resolution behavior.</param>
```
Comment (azabbasi) [inline sdk/modelsrepository/Azure.IoT.ModelsRepository/src/ModelsRepositoryClient.cs:74]: nice reword :)

File: sdk/modelsrepository/Azure.IoT.ModelsRepository/src/ModelsRepositoryClientOptions.cs [hunk 2]
```diff
@@ -39,18 +39,10 @@ public enum ServiceVersion
         /// The <see cref="ServiceVersion"/> of the service API used when
         /// making requests.
         /// </param>
-        /// <param name="dependencyResolution">The model dependency resolution options.</param>
         public ModelsRepositoryClientOptions(
-            ServiceVersion version = LatestVersion,
-            ModelDependencyResolution dependencyResolution = ModelDependencyResolution.Enabled)
+            ServiceVersion version = LatestVersion)
```
Comment (azabbasi) [inline sdk/modelsrepository/Azure.IoT.ModelsRepository/src/ModelsRepositoryClientOptions.cs:43]: I really like this change.

File: sdk/modelsrepository/Azure.IoT.ModelsRepository/src/ModelsRepositoryMetadata.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        [JsonPropertyName("features")]
+        public RepositoryFeatures Features { get; set; }
+
+        internal class RepositoryFeatures
+        {
+            [JsonPropertyName("expanded")]
+            public bool Expanded { get; set; }
+            [JsonPropertyName("index")]
+            public bool Index { get; set; }
+        }
```
Comment (azabbasi) [inline sdk/modelsrepository/Azure.IoT.ModelsRepository/src/ModelsRepositoryMetadata.cs:32]: nit: needs new line.

File: sdk/modelsrepository/Azure.IoT.ModelsRepository/src/ModelsRepositoryMetadata.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        [JsonPropertyName("totalModelCount")]
+        public int TotalModelCount { get; set; }
+        [JsonPropertyName("features")]
+        public RepositoryFeatures Features { get; set; }
+
+        internal class RepositoryFeatures
+        {
+            [JsonPropertyName("expanded")]
+            public bool Expanded { get; set; }
+            [JsonPropertyName("index")]
```
Comment (azabbasi) [inline sdk/modelsrepository/Azure.IoT.ModelsRepository/src/ModelsRepositoryMetadata.cs:30]: nit: I personally am in favor of new lines between properties.

Comment (azabbasi) [reply sdk/modelsrepository/Azure.IoT.ModelsRepository/src/ModelsRepositoryMetadata.cs:30]: it helps with readability

Comment (digimaun) [reply sdk/modelsrepository/Azure.IoT.ModelsRepository/src/ModelsRepositoryMetadata.cs:30]: Will make the change

**Review by azabbasi — APPROVED** [review]
LGTM

**Pull request merged**